### PR TITLE
enhance/signals-graph-preview

### PR DESCRIPTION
**Pull request by haritonasty**
**Summary**
- red dots instead green line for  `Price` and `DAA` signals 
- tooltip with fixed position
- gradient under line
- horizontal helpers

**In next PR**
- limit line

**Screenshots**
image
image

### CODE REVIEW

File: src/ducks/Signals/chart/VisualBacktestChart.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+          allowDataOverflow
+          domain={['dataMin', 'dataMax']}
+          hide
+        />
+
+        <YAxis hide domain={['auto', 'dataMax']} dataKey={dataKeys.metric} />
+
+        {markup}
+
+        {triggeredSignals.map((point, i) => {
```
Comment (Partysun) [inline src/ducks/Signals/chart/VisualBacktestChart.js]: don't use one letter variable name.

File: src/ducks/Signals/chart/SignalPreview.js [hunk 2]
```diff
@@ -42,56 +32,35 @@ const SignalPreviewChart = ({
     return acc
   }, {})
 
+  const metricsModified = metrics.map(name =>
```
Comment (Partysun) [inline src/ducks/Signals/chart/SignalPreview.js]: Possible, it's the too generic name for this.

File: src/ducks/Signals/chart/VisualBacktestChart.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+import { getDateFormats, getTimeFormats } from '../../../utils/dates'
+import chartStyles from './../../SANCharts/Chart.module.scss'
+import sharedStyles from './../../SANCharts/ChartPage.module.scss'
+import styles from './SignalPreview.module.scss'
+
+export const getDataKeys = (signal = {}) => {
+  if (signal.active_addresses) {
+    return { metric: 'dailyActiveAddresses', signal: 'active_addresses' }
+  }
+  if (signal.price) return { metric: 'priceUsd', signal: 'price' }
```
Comment (Partysun) [inline src/ducks/Signals/chart/VisualBacktestChart.js]: Each time when I see "if" and return without brackets, I am feeling possible error. But here I see an inconsistency, 2 "if's" one with brackets, one is without

File: src/ducks/Signals/chart/VisualBacktestChart.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+const mapWithTimeseries = items =>
+  items.map(item => ({ ...item, datetime: +new Date(item.datetime) }))
+
+const mapWithMidnightTime = date => {
+  const datetime = new Date(date)
+  datetime.setUTCHours(0, 0, 0, 0)
+  return +new Date(datetime)
+}
+
+const VisualBacktestChart = ({
```
Comment (Partysun) [inline src/ducks/Signals/chart/VisualBacktestChart.js]: We can move VisualBacktestChart upper and 3 functions below the main func.

File: src/ducks/Signals/chart/VisualBacktestChart.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+export const getDataKeys = (signal = {}) => {
+  if (signal.active_addresses) {
+    return { metric: 'dailyActiveAddresses', signal: 'active_addresses' }
+  }
+  if (signal.price) return { metric: 'priceUsd', signal: 'price' }
+  return { metric: 'priceUsd' }
+}
+
+const mapWithTimeseries = items =>
+  items.map(item => ({ ...item, datetime: +new Date(item.datetime) }))
```
Comment (Partysun) [inline src/ducks/Signals/chart/VisualBacktestChart.js]: We do it in so many places. Maybe we can do it in GetTimeSeries during fetching data. But it's just thought.

File: src/ducks/Signals/chart/VisualBacktestChart.js [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+          if (isPrice) {
+            const item = data.find(({ datetime }) => datetime === date)
+            price = item ? item.priceUsd : point.price
+          }
+
+          return dataKeys.signal ? (
+            <ReferenceDot
+              x={date}
+              y={price || point[dataKeys.signal]}
```
Comment (Partysun) [inline src/ducks/Signals/chart/VisualBacktestChart.js]: If you make the function, which will return y, you can combine 3 places from the code to one and avoid "let"

File: src/ducks/Signals/chart/VisualBacktestChart.js (hunk 5 above)
Comment (haritonasty) [reply src/ducks/Signals/chart/VisualBacktestChart.js]: l'll investigate

**Pull request merged**